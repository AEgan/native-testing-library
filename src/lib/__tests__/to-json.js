import React from 'react';
import { Text, View } from 'react-native';

import { cleanup, prettyPrint, render, toJSON } from '../../';

afterEach(cleanup);

test('it converts to json', () => {
  function ParentComponent({ children }) {
    return <View>{children}</View>;
  }

  function MiddleComponent({ children }) {
    return (
      <View>
        <Text>{children}</Text>
      </View>
    );
  }

  const { baseElement, container } = render(
    <ParentComponent>
      <View>
        <MiddleComponent>hello</MiddleComponent>
        <View>
          <Text onPress={jest.fn()}>world</Text>
          <Text>foo bar</Text>
        </View>
        <View />
      </View>
    </ParentComponent>,
  );

  expect(prettyPrint(toJSON(container))).toMatchSnapshot();
  expect(prettyPrint(toJSON(baseElement))).toMatchSnapshot();
});
