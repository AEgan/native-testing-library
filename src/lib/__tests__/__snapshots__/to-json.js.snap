// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it converts to json 1`] = `
"[36m<View[39m
  [33mtestID[39m=[32m\\"ntl-container\\"[39m
[36m>[39m
  [36m<View>[39m
    [36m<View>[39m
      [36m<View>[39m
        [36m<Text>[39m
          [0mhello[0m
        [36m</Text>[39m
      [36m</View>[39m
      [36m<View>[39m
        [36m<Text>[39m
          [0mworld[0m
        [36m</Text>[39m
        [36m<Text>[39m
          [0mfoo bar[0m
        [36m</Text>[39m
      [36m</View>[39m
      [36m<View />[39m
    [36m</View>[39m
  [36m</View>[39m
[36m</View>[39m"
`;

exports[`it converts to json 2`] = `
"[36m<View[39m
  [33mpointerEvents[39m=[32m\\"box-none\\"[39m
  [33mstyle[39m=[32m{
    Object {
      \\"flex\\": 1,
    }
  }[39m
[36m>[39m
  [36m<View[39m
    [33mcollapsable[39m=[32m{true}[39m
    [33mpointerEvents[39m=[32m\\"box-none\\"[39m
    [33mstyle[39m=[32m{
      Object {
        \\"flex\\": 1,
      }
    }[39m
  [36m>[39m
    [36m<View[39m
      [33mtestID[39m=[32m\\"ntl-container\\"[39m
    [36m>[39m
      [36m<View>[39m
        [36m<View>[39m
          [36m<View>[39m
            [36m<Text>[39m
              [0mhello[0m
            [36m</Text>[39m
          [36m</View>[39m
          [36m<View>[39m
            [36m<Text>[39m
              [0mworld[0m
            [36m</Text>[39m
            [36m<Text>[39m
              [0mfoo bar[0m
            [36m</Text>[39m
          [36m</View>[39m
          [36m<View />[39m
        [36m</View>[39m
      [36m</View>[39m
    [36m</View>[39m
  [36m</View>[39m
[36m</View>[39m"
`;
